cinema/views.py

```diff
@@ -1 +1,67 @@
-# write views here
+from rest_framework import viewsets
+from .models import (
+    CinemaHall,
+    Genre,
+    Actor,
+    Movie,
+    MovieSession,
+    Order,
+    Ticket,
+)
+from .serializers import (
+    GenreSerializer,
+    ActorSerializer,
+    MovieSerializer,
+    MovieListSerializer,
+    MovieRetrieveSerializer,
+    CinemaHallSerializer,
+    MovieSessionSerializer,
+    MovieSessionRetrieveSerializer, MovieSessionListSerializer
+)
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all()
+
+    def get_queryset(self):
+        if self.action in ["list", "retrieve"]:
+            return self.queryset.prefetch_related("genres", "actors")
+        return self.queryset
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieListSerializer
+        elif self.action == "retrieve":
+            return MovieRetrieveSerializer
+        return MovieSerializer
+
+
+class MovieSessionViewSet(viewsets.ModelViewSet):
+    queryset = MovieSession.objects.all()
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieSessionListSerializer
+        elif self.action == "retrieve":
+            return MovieSessionRetrieveSerializer
+        return MovieSessionSerializer
+
+    def get_queryset(self):
+        if self.action in ["list", "retrieve"]:
+            return self.queryset.select_related()
```

Comment:
The `select_related()` method in `get_queryset` of `MovieSessionViewSet` should specify the related fields to optimize database queries. Consider using `select_related('movie', 'cinema_hall')` to fetch related `Movie` and `CinemaHall` objects in a single query.